.github/workflows/codeql-analysis.yml

```diff
@@ -28,7 +28,7 @@ jobs:
 
     steps:
     - name: Checkout repository
-      uses: actions/checkout@v2
+      uses: actions/checkout@v4
 
     - name: Initialize CodeQL
       uses: github/codeql-action/init@v2
```

Comment:
can we update all actions while were doing these updates to make sure we're not falling behind?